### PR TITLE
Transpose csv to facilitate reading

### PR DESCRIPTION
**Pull request by lrouhier**
table to compare ivadomed with other project is now vertical to facilitate reading.

### CODE REVIEW

**Comment by lrouhier** [comment]
@jcohenadad is this what you had in mind ? https://ivado-medical-imaging--567.org.readthedocs.build/en/567/purpose.html

**Comment by charleygros** [comment]
@lrouhier : I like the transposed version.
Would it be possible to have:
```
Classification
Segmentation
Detection
```
instead of all in one row? I am just hoping to make the table more compact.

**Comment by lrouhier** [comment]
Ok sther will be Classification
Segmentation
Detection
Generation
clustering
Generation 
is this what you thinking about ?

File: docs/source/purpose.rst [hunk 1]
```diff
@@ -26,32 +26,11 @@ of the existing projects. This table was mostly based on the existing documentat
 understand that the field is rapidly evolving, and that this table might reflect the reality. If you notice
 inconsistencies, please let us know by `opening an issue <https://github.com/ivadomed/ivadomed/issues>`_.
 
-General overview
-++++++++++++++++
-
 .. csv-table::
-   :file: general_overview.csv
+   :file: comparison_dl_soft_transposed.csv
```
Comment (jcohenadad) [inline docs/source/purpose.rst:30]: i would simply call the table the way it was before, ie: "comparison_other_projects_table.csv"

File: docs/source/comparison_dl_soft_transposed.csv [hunk 2]
```diff
@@ -0,0 +1,23 @@
+**Name**,`ivadomed <https://github.com/ivadomed/ivadomed>`_,`monai <https://github.com/Project-MONAI/MONAI>`_,`delira <https://github.com/justusschock/delira>`_,`MIC-DKFZ <https://github.com/MIC-DKFZ/medicaldetectiontoolkit>`_,`ANTsPyNet <https://github.com/ANTsX/ANTsPyNet>`_,`DLTK <https://github.com/DLTK/DLTK>`_,`MIScnn <https://github.com/frankkramer-lab/MIScnn>`_,`niftytorch <https://niftytorch.github.io/doc/>`_,`DeepNeuro <https://github.com/QTIM-Lab/DeepNeuro>`_
+**BIDS**,|yes|,|no|,|no|,|no|,|no|,|no|,|no|,|yes|,|no|
+**DL base library**,PyTorch,PyTorch,PyTorch and TensorFlow,PyTorch,Tensorflow/Keras,Tensorflow,Tensorflow/Keras,PyTorch,Tensorflow/Keras
+"**Task (segmentation, detection, classification)**","Classification, Segmentation, Detection","Segmentation, Classification","Classification, Generation, Segmentation",Detection,"Classification, Segmentation, Clustering, Registration","Classification, Segmentation,Registration",Segmentation,"Classification, Segmentation ",Segmentation
```
Comment (jcohenadad) [inline docs/source/comparison_dl_soft_transposed.csv:4]: this row is to verbose, and to echo @charleygros in https://github.com/ivadomed/ivadomed/pull/567#issuecomment-741329078, i suggest we squeeze it. My suggestion would be:
~~~
"**Task (*)**","Class, Seg, Detect","Class, Seg"
~~~
and below the table, have a legend:
(*): Class: Classification | Detect: Detection | Seg: Segmentation

**Comment by jcohenadad** [comment]
@lrouhier i think we still want to transpose it, right?

image

**Comment by lrouhier** [comment]
Ah sorry, I put the version before the transpose

**Comment by jcohenadad** [comment]
?

image

**Comment by lrouhier** [comment]
I don't know why you are seeing the version merged with master, but here I have the right one: https://ivado-medical-imaging--567.org.readthedocs.build/en/567/purpose.html

**Review by jcohenadad — CHANGES_REQUESTED** [review]
TensorFlow -> TF

File: docs/source/purpose.rst (hunk 1 above)
Comment (lrouhier) [reply docs/source/purpose.rst]: Done

File: docs/source/comparison_dl_soft_transposed.csv (hunk 2 above)
Comment (lrouhier) [reply docs/source/comparison_dl_soft_transposed.csv]: Done

**Comment by lrouhier** [comment]
> TensorFlow -> TF

Done

**Comment by jcohenadad** [comment]
hum, weird... am i seeing the latest version?
image

**Comment by lrouhier** [comment]
sorry about that, this time it's right

File: docs/source/comparison_other_projects_table.csv [hunk 3]
```diff
@@ -1,14 +1,30 @@
...
+**Transfer Learning**,|yes|,|no|,|yes|,|no|,|no|,|yes|,|no|,|no|,|no|
+**Pre-processing tools**,|yes|,|yes|,|no|,|no|,|no|,|no|,|yes|,|no|,|yes|
+**Post-processing tools**,|yes|,|no|,|no|,|no|,|no|,|no|,|yes|,|no|,|yes|
+**User case examples**,|yes|,|yes|,|yes|,|no|,|yes| ,|yes|,|no|,|yes|,|yes|
+**Multi-GPU data parallelism**,|no|,|yes|,|yes|,|no|,|no|,|no|,|yes|,|yes|,|no|
+**Automatic Model evaluation**,|yes|,|yes|,|no|,|yes|,|no|,|no|,|yes|,|no|,|no|
+**Input region of interest**,|yes| ,|yes| ,|no|,|yes|,|no|,|no|,|no|,|yes|,|no|
+**Missing modality**,|yes|,|no|,|no|,|no|,|no|,|no|,|no|,|no|,|no|
+**Model performance comparison**,|yes|,|no|,|no|,|no|,|no|,|no|,|no|,|no|,|no|
+**Automatic hyperparameter optimisation**,|yes|,|no|,|no|,|no|,|no|,|no|,|no|,|yes|,|no|
```
Comment (jcohenadad) [inline docs/source/comparison_other_projects_table.csv:29]: To save space
```suggestion
**Hyperparam optimisation**,|yes|,|no|,|no|,|no|,|no|,|no|,|no|,|yes|,|no|
```

File: docs/source/comparison_other_projects_table.csv [hunk 4]
```diff
@@ -1,14 +1,30 @@
...
+**Uncertainty**,|yes|,|no| ,|no|,|no|,|no|,|no| ,|no| ,|no| ,|no|
+**Transfer Learning**,|yes|,|no|,|yes|,|no|,|no|,|yes|,|no|,|no|,|no|
+**Pre-processing tools**,|yes|,|yes|,|no|,|no|,|no|,|no|,|yes|,|no|,|yes|
+**Post-processing tools**,|yes|,|no|,|no|,|no|,|no|,|no|,|yes|,|no|,|yes|
+**User case examples**,|yes|,|yes|,|yes|,|no|,|yes| ,|yes|,|no|,|yes|,|yes|
+**Multi-GPU data parallelism**,|no|,|yes|,|yes|,|no|,|no|,|no|,|yes|,|yes|,|no|
+**Automatic Model evaluation**,|yes|,|yes|,|no|,|yes|,|no|,|no|,|yes|,|no|,|no|
+**Input region of interest**,|yes| ,|yes| ,|no|,|yes|,|no|,|no|,|no|,|yes|,|no|
+**Missing modality**,|yes|,|no|,|no|,|no|,|no|,|no|,|no|,|no|,|no|
+**Model performance comparison**,|yes|,|no|,|no|,|no|,|no|,|no|,|no|,|no|,|no|
```
Comment (jcohenadad) [inline docs/source/comparison_other_projects_table.csv:28]: to save space
```suggestion
**Model comparison**,|yes|,|no|,|no|,|no|,|no|,|no|,|no|,|no|,|no|
```

File: docs/source/comparison_other_projects_table.csv [hunk 5]
```diff
@@ -1,14 +1,30 @@
...
-DeepNeuro,https://github.com/QTIM-Lab/DeepNeuro,|no|,Tensorflow/Keras,Segmentation,"2D, 3D",|no| ,|no|,|no|,|no|,|yes|,|yes|,|yes|,|no|,|no|,|no|,|no|,|no|,|no|,|yes|
\ No newline at end of file
+",|no| ,|no| ,|yes| ,|yes| ,|yes| ,|no| ,|no|
+**Uncertainty**,|yes|,|no| ,|no|,|no|,|no|,|no| ,|no| ,|no| ,|no|
+**Transfer Learning**,|yes|,|no|,|yes|,|no|,|no|,|yes|,|no|,|no|,|no|
+**Pre-processing tools**,|yes|,|yes|,|no|,|no|,|no|,|no|,|yes|,|no|,|yes|
+**Post-processing tools**,|yes|,|no|,|no|,|no|,|no|,|no|,|yes|,|no|,|yes|
+**User case examples**,|yes|,|yes|,|yes|,|no|,|yes| ,|yes|,|no|,|yes|,|yes|
+**Multi-GPU data parallelism**,|no|,|yes|,|yes|,|no|,|no|,|no|,|yes|,|yes|,|no|
+**Automatic Model evaluation**,|yes|,|yes|,|no|,|yes|,|no|,|no|,|yes|,|no|,|no|
```
Comment (jcohenadad) [inline docs/source/comparison_other_projects_table.csv:25]:
```suggestion
**Model evaluation**,|yes|,|yes|,|no|,|yes|,|no|,|no|,|yes|,|no|,|no|
```

File: docs/source/comparison_other_projects_table.csv (hunk 4 above)
Comment (jcohenadad) [reply docs/source/comparison_other_projects_table.csv:28]: also, i would move this right below "Model evaluation"

File: docs/source/comparison_other_projects_table.csv [hunk 6]
```diff
@@ -1,14 +1,30 @@
...
+**Input region of interest**,|yes| ,|yes| ,|no|,|yes|,|no|,|no|,|no|,|yes|,|no|
+**Missing modality**,|yes|,|no|,|no|,|no|,|no|,|no|,|no|,|no|,|no|
+**Model performance comparison**,|yes|,|no|,|no|,|no|,|no|,|no|,|no|,|no|,|no|
+**Automatic hyperparameter optimisation**,|yes|,|no|,|no|,|no|,|no|,|no|,|no|,|yes|,|no|
+**Packaged Multi-center Model**,|yes|,|no|,|no|,|no|,|no|,|no|,|no|,|no|,|yes|
```
Comment (jcohenadad) [inline docs/source/comparison_other_projects_table.csv:30]:
```suggestion
**Multi-center models**,|yes|,|no|,|no|,|no|,|no|,|no|,|no|,|no|,|yes|
```

File: docs/source/comparison_other_projects_table.csv [hunk 7]
```diff
@@ -1,14 +1,30 @@
...
-**ivadomed**,https://github.com/ivadomed/ivadomed,|yes|,PyTorch,"Classification, Segmentation, Detection","2D, 3D",|yes| ,|yes| ,|yes|,|yes|,|yes|,|yes|,|yes|,|no|,|yes|,|yes| ,|yes|,|yes|,|yes|,|yes|
-monai,https://github.com/Project-MONAI/MONAI,|no|,PyTorch,"Segmentation, Classification","2D, 3D",|yes| ," 
+**Name**,`ivadomed <https://github.com/ivadomed/ivadomed>`_,`monai <https://github.com/Project-MONAI/MONAI>`_,`delira <https://github.com/justusschock/delira>`_,`MIC-DKFZ <https://github.com/MIC-DKFZ/medicaldetectiontoolkit>`_,`ANTsPyNet <https://github.com/ANTsX/ANTsPyNet>`_,`DLTK <https://github.com/DLTK/DLTK>`_,`MIScnn <https://github.com/frankkramer-lab/MIScnn>`_,`niftytorch <https://niftytorch.github.io/doc/>`_,`DeepNeuro <https://github.com/QTIM-Lab/DeepNeuro>`_
+**BIDS (1)**,|yes|,|no|,|no|,|no|,|no|,|no|,|no|,|yes|,|no|
+**DL base library**,PyTorch,PyTorch,PyTorch and TF,PyTorch,TF/Keras,TF,TF/Keras,PyTorch,TF/Keras
```
Comment (jcohenadad) [inline docs/source/comparison_other_projects_table.csv:3]: to be consistent with other cells
```suggestion
**DL base library**,PyTorch,PyTorch,"PyTorch, TF",PyTorch,TF/Keras,TF,TF/Keras,PyTorch,TF/Keras
```

**Review by jcohenadad — COMMENTED** [review]
can you generate a table where all values are centered in the cells (currently only the checks/cross are centered)-- thanks

**Comment by lrouhier** [comment]
I am not sure I can (because a command needs to go inside the file and I haven't found a way to do this). 
* Creating an html command (as seen currently in purpose.rst with | cent | does not seem to work)
* Creatin a css class to center the text does not seems to work either. 

@kousu do you have any idea how to do this. 

I also added the link of the original table on the drive as a comment in purpose.rst.

**Review by jcohenadad — CHANGES_REQUESTED** [review]
i'll fix the legend 1,2 --> need to be more verbose-- i'll do when i find a minute

**Comment by kousu** [comment]
> I am not sure I can (because a command needs to go inside the file and I haven't found a way to do this).
> 
>     * Creating an html command (as seen currently in purpose.rst with | cent | does not seem to work)
> 
>     * Creatin a css class to center the text does not seems to work either.
> 
> 
> @kousu do you have any idea how to do this.
> 
> I also added the link of the original table on the drive as a comment in purpose.rst.

I'll look at this in a bit.

File: docs/source/purpose.rst [hunk 8]
```diff
@@ -26,31 +30,15 @@ of the existing projects. This table was mostly based on the existing documentat
...
-   :file: model_evaluation.csv
+   :file: comparison_other_projects_table.csv
 
-Documentation
-+++++++++++++
+(1): `Brain Imaging Data Structure <https://bids.neuroimaging.io/>`_ 
 
-.. csv-table:: 
-   :file: documentation.csv
+(2): Class: Classification | Seg: Segmentation | Detect: Detection | Gen: Generation | Clust: Clustering | Reg: Registration
```
Comment (jcohenadad) [inline docs/source/purpose.rst:41]:
```suggestion
(2): **Class**: Classification | **Seg**: Segmentation | **Detect**: Detection | **Gen**: Generation | **Clust**: Clustering | **Reg**: Registration
```

File: docs/source/purpose.rst [hunk 9]
```diff
@@ -26,32 +30,14 @@ of the existing projects. This table was mostly based on the existing documentat
...
-   :file: model_evaluation.csv
-
-Documentation
-+++++++++++++
-
-.. csv-table:: 
-   :file: documentation.csv
+   :file: comparison_other_projects_table.csv
 
+(1): `"BIDS" stands for the Brain Imaging Data Structure <https://bids.neuroimaging.io/>`_, which is a convention initiated by the neuroimaging community to organize datasets (filenames, metadata, etc.). This facilitates the sharing of datasets and minimizes the burden of organizing datasets for training.
```
Comment (jcohenadad) [inline docs/source/purpose.rst:39]:
```suggestion
(1): "BIDS" stands for the `Brain Imaging Data Structure <https://bids.neuroimaging.io/>`_, which is a convention initiated by the neuroimaging community to organize datasets (filenames, metadata, etc.). This facilitates the sharing of datasets and minimizes the burden of organizing datasets for training.
```

**Review by jcohenadad — APPROVED** [review]
Thank you @lrouhier 🏅

**Pull request merged**